modules/javafx.controls/src/main/java/javafx/scene/control/skin/TreeTableRowSkin.java

```diff
@@ -332,7 +331,7 @@ public final DoubleProperty indentProperty() {
 
     private void updateTreeItem() {
         unregisterInvalidationListeners(graphicProperty());
-        treeItem = getSkinnable().getTreeItem();
+        treeItem = (getSkinnable() == null) ? null : getSkinnable().getTreeItem();
```

Comment:
This is not needed if the weak listener change is reverted in the constructor. The cause for it being `null` is that the listener is now no longer removed by `dispose` where before (with `ListenerHelper` or `regsisterXXXListener`) it would have been.

Also, I think when `dispose` is called, this skin should still do its utmost best to remove all listeners immediately.

Reply:
The problem is, `dispose()` is never called for cells discarded by the refresh() method.

In the absence of explicit release, these cells are still listening - so using a weak listener in this case is, I think, the right approach.  Adding explicit release per your (absolutely valid) suggestion is a good idea, but much, much more complicated task.

Reply:
Even if you are using weak listeners, `dispose` should remove them.

Reply:
revered back to using register*Listener, which creates a weak listener and also disconnects them in dispose()